### PR TITLE
fix: get os from context tool

### PR DESCRIPTION
**Pull request by njhale**
Get the local operating system using the official context tool and remove instructions to strip the file suffix from input. The latter will be handled in `gptscript` when the `--ui` option is provided.

Depends on https://github.com/gptscript-ai/gptscript/pull/530

### CODE REVIEW

File: tool.gpt [hunk 1]
```diff
@@ -1,21 +1,16 @@
...
+
+Never run steps in parallel.
+
+Perform the following steps in the order they appear:
+1. Run the service
+2. Get the port for the running service
+3. If no {file} was provided AND the local operating system is windows, use the open-windows tool then skip the remaining steps
+4. If no {file} was provided use the open-nix tool then skip the remaining steps
+5. If the operating system is windows, use the open-file-windows tool then skip the remaining steps
+6. Use the open-file-nix tool
```
Comment (njhale) [inline tool.gpt:13]: I think there's probably a more clever way to select the right tools based on OS via contexts, but ATM this was the most straightforward way I could come up with.

File: tool.gpt [hunk 2]
```diff
@@ -1,21 +1,16 @@
...
-        - If a file or URL is provided, use the start file tool.
-        - If no file or URL is provided, use the start tool.
-    - If the operating system is not Windows:
-        - If a file or URL is provided, use the open-file tool.
-        - If a file or URL is not provided, use the open tool.
+context: github.com/gptscript-ai/context/os
+tools: service, port, open-file-nix, open-nix, open-file-windows, open-windows
+params: file: (optional) A reference to a tool file. Can refer to either a local or remote file.
+
+Never run steps in parallel.
```
Comment (njhale) [inline tool.gpt:5]: Was having a problem with steps 1 and 2 running in parallel. This corrects for that.

File: tool.gpt [hunk 3]
```diff
@@ -61,7 +46,7 @@ open http://localhost:${PORT}
 
 ---
 
-name: start-file
+name: open-file-windows
```
Comment (njhale) [inline tool.gpt:49]: renamed to make it more obvious that these toosl are just the "windows variants" of the others.

Comment (thedadams) [reply tool.gpt:49]: Eh, OK.

File: tool.gpt (hunk 1 above)
Comment (thedadams) [reply tool.gpt:13]: I'm a little surprised that the LLM does the right thing and doesn't try to do _something_ for each listed step.

**Review by thedadams — APPROVED** [review]

File: tool.gpt (hunk 1 above)
Comment (njhale) [reply tool.gpt:13]: I did some basic testing to make sure it works for remote and local paths, but I didn't try every combination here (e.g. on windows). I'll check on windows and take a look at the events to see exactly what the LLM does here.

File: tool.gpt (hunk 3 above)
Comment (njhale) [reply tool.gpt:49]: Well, I can switch it back if we're feeling "meh" about it. Lmk

**Review by tylerslaton — APPROVED** [review]

File: tool.gpt (hunk 1 above)
Comment (njhale) [reply tool.gpt:13]: checked and remote URLs are working properly on windows

**Pull request merged**